src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -169,7 +169,24 @@ private static TemplateMatcher GetInstanceRaiseEventRoute()
             Stopwatch stopwatch = Stopwatch.StartNew();
             while (true)
             {
-                DurableOrchestrationStatus status = await client.GetStatusAsync(instanceId);
+                DurableOrchestrationStatus status = null;
+
+                if (client is DurableClient durableClient && durableClient.DurabilityProvider.SupportsPollFreeWait)
+                {
+                    try
+                    {
+                        var state = await durableClient.DurabilityProvider.WaitForOrchestrationAsync(instanceId, null, timeout, CancellationToken.None);
+                        status = DurableClient.ConvertOrchestrationStateToStatus(state);
+                    }
+                    catch (TimeoutException)
+                    {
+                    }
+                }
+                else
+                {
+                    status = await client.GetStatusAsync(instanceId);
+                }
+

```

Comment:
Can we document / comment about the difference between these two execution paths?

Reply:
Yes.